tempesta_fw/http.c

```diff
@@ -683,6 +683,36 @@ tfw_http_add_hdr_via(TfwHttpMsg *hm)
 }
 
 static int
+tfw_http_add_hdr_110(TfwHttpMsg *hm)
+{
+	int r;
+	TfwVhost *vhost = tfw_vhost_get_default();
+	unsigned int vlen = SLEN("Response is stale. : ");
+
+	TfwStr rh = {
+#define S_Stale	"Warning: 110: "
+		.ptr = (TfwStr []) {
+			{ .ptr = S_Stale, .len = SLEN(S_Stale) },
+			{ .ptr = (void *)"Response is stale. : ",
+			  .len = vlen },
+			{ .ptr = *this_cpu_ptr(&g_buf),
+			  .len = vhost->hdr_via_len },
+		},
+		.len = SLEN(S_Stale) + vlen + vhost->hdr_via_len,
+		.eolen = 2,
+		.flags = 3 << TFW_STR_CN_SHIFT
+#undef S_Stale
+	};
+
+	r = tfw_http_msg_hdr_add(hm, &rh);
```

Comment:
All of the above is completely bogus. `vhost->hdr_via_len` is completely unrelated to the `Warning` header field. Please use `tfw_http_msg_hdr_xfrm()` to add the `Warning` header field to a response.